### DISCUSSION TITLE
chromium based browser loses focus when activate

### DISCUSSION THREAD

**Discussion by aviadHCL** [cefsharp/CefSharp#4322] [General]
Hi, I've recently updated from Cef version 100.0.230 to 107.1.90 and when I open the chromium-based browser now and clicking on "alt"+"f4" it won't respond until I'll click on the inside of the opened window. Is this a bug? known one? should it be reported?
it reproduced for me on any version from 104 onward

**Comment by amaitland** [comment]
> known one? should it be reported?

CEF has it's own issue tracker on bit bucket. You need to search there before opening an issue. This is likely an upstream CEF issue, there were focus changes in CEF to fix some IME related issues in this version if I remember correctly.

Likely the same as https://bitbucket.org/chromiumembedded/cef/issues/3426/clicking-and-dragging-the-cef-window

> clicking on "alt"+"f4" 

No idea what you mean by this,

**Reply by aviadHCL** [reply]
@amaitland 

`> No idea what you mean by this,`
I mean when sending keys alt+f4 the opened chrome window doesn't close as it's losing focus

**Reply by amaitland** [reply]
The relevant upstream commits are:

- https://bitbucket.org/chromiumembedded/cef/commits/2587cf23c5110bac3488ce7f2c98f6e08722aa50
- https://bitbucket.org/chromiumembedded/cef/commits/47dab09c5e36a6e293c98776013659da7d45ec83



> Likely the same as https://bitbucket.org/chromiumembedded/cef/issues/3426/clicking-and-dragging-the-cef-window

I'd say it's related to this issue. You can however open your own issue on the [CEF ISSUE TRACKER](https://bitbucket.org/chromiumembedded/cef/issues/new) if you feel it's different. You'll need to provide a detailed list of steps to reproduce the problem. The details you've provided here are too vague IMHO.

https://bitbucket.org/chromiumembedded/cef/issues/new

**Reply by amaitland** [reply]
Looks like you created https://bitbucket.org/chromiumembedded/cef/issues/3436/chromium-based-browser-loses-focus-when

> chromium based browser loses focus when activate

How did you determine the browser loses focus? Spy++?

The `ChromiumWebBrowser` instance is focused as expected, after the `CefBrowser` instance is created we call `CefBrowserHost::SetFocus(true);` to focus the `CefBrowser` instance. This no longer appears to actually give the browser focus, I'd say it's a timing issue. 

A quick look and I think we should be able to implement a workaround until the `upstream` issue is fixed, let me know if your company would like to [Sponsor](https://github.com/sponsors/amaitland) the project and I'll prioritise looking into a workaround.

**Reply by amaitland** [reply]
Version `109.1.110` includes a workaround and is now available.
